src/main/java/com/dnastack/ga4gh/search/adapter/presto/PrestoTableDataModelSource.java

```diff
@@ -0,0 +1,97 @@
+package com.dnastack.ga4gh.search.adapter.presto;
+
+import com.dnastack.ga4gh.search.client.indexingservice.IndexingServiceClient;
+import com.dnastack.ga4gh.search.client.indexingservice.model.LibraryItem;
+import com.dnastack.ga4gh.search.client.tablesregistry.TablesRegistryClient;
+import com.dnastack.ga4gh.search.client.tablesregistry.TablesRegistryClientConfig;
+import com.dnastack.ga4gh.search.client.tablesregistry.model.ListTableRegistryEntry;
+import com.dnastack.ga4gh.search.model.DataModel;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PrestoTableDataModelSource {
+    @Autowired(required = false)
+    private DataModelSource dataModelSource;
+
+    public DataModel getDataModel(String tableId) {
+        if (this.dataModelSource == null) {
+            return null;
+        }
+        return this.dataModelSource.getDataModel(tableId);
+    }
+
+    @Bean
+    @ConditionalOnExpression("'${app.presto.tables.data-model-source}' == 'indexing-service'")
+    private DataModelSource indexingServiceDataModelSource() {
+        return new IndexingServiceDataModelSource();
+    }
+
+    @Bean
+    @ConditionalOnExpression("'${app.presto.tables.data-model-source}' == 'tables-registry'")
+    private DataModelSource tablesRegistryServiceDataModelSource() {
+        return new TablesRegistryServiceDataModelSource();
+    }
+
+    private interface DataModelSource {
+        DataModel getDataModel(String tableId);
+    }
+
+    @Slf4j
+    private static class IndexingServiceDataModelSource implements DataModelSource {
+        @Autowired
+        private IndexingServiceClient indexingServiceClient;
+        private final ObjectMapper objectMapper = new ObjectMapper()
+                .setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE)
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
+        @Override
+        public DataModel getDataModel(String tableId) {
```

Comment:
Account for three part table Id and extract last two parts to send to indexing service

Reply:
Refer to [data lake connector spec](https://github.com/DNAstack/data-lake-connector/blob/aa8f93d1fb0eb25a25572faf362a9c4cdd446e81/api-spec.yml#L714)